osu.Game/Utils/HumanizerUtils.cs

```diff
@@ -26,5 +26,32 @@ public static string Humanize(DateTimeOffset input)
                 return input.Humanize(culture: new CultureInfo("en-US"));
             }
         }
+
+        /// <summary>
+        /// Turns the current or provided big number into a readable string
+        /// </summary>
+        /// <param name="input">The number to be humanized</param>
+        /// <returns>simplified number with a suffix</returns>
+        public static string Humanize(long input)
+        {
+            const int k = 1000;
+
+            var suffixes = new[]
+            {
+                "",
+                "k",
+                "million",
+                "billion",
+                "trillion",
+                "quadrillion",
+                "quintillion",
+            };
+
+            if (input < k)
+                return input.ToString();
+
+            int i = (int)Math.Floor(Math.Round(Math.Log(input, k)));
```

Comment:
This rounds the edge case (3n-1)-digit numbers up now:

![image](https://user-images.githubusercontent.com/20418176/65599688-aa823e80-df9e-11e9-89a2-89373df333f8.png)

When I suggested checking I actually wanted to see if it would round *down* correctly for large values. Looking at what `Math.Log` does, it unfortunately doesn't handle large bases and large numbers well in that case:

```csharp
> Math.Round(Math.Log(999_999_999_999_999, 1000))
5
> Math.Log(1_000_000_000_000_000, 1000)
5
> Math.Log(1_000_000_000_000_000, 1000) - Math.Log(999_999_999_999_999, 1000)
0
```

Not sure what kind of behaviour we want to get here from a UX standpoint.

Reply:
Since the built-in `Math` methods are broken here, a temporary fix can be used I suppose, mine would be checking if `Math.Pow(k, i)` is bigger than the giving input and fix that by looping till it's less:
```csharp
while (Math.Pow(k, i) > input) i--;
```
with a note why this line is there.